src/main/histFileMod.F90

```diff
@@ -4733,224 +4824,234 @@ subroutine hist_restart_ncd (bounds, ncid, flag, rdate)
     !================================================
 
        call ncd_inqdlen(ncid,dimid,ntapes_onfile, name='ntapes')
-       if (is_restart()) then
+       if_restart1: if (is_restart()) then
           if (ntapes_onfile /= ntapes) then
              write(iulog,*) 'ntapes = ', ntapes, ' ntapes_onfile = ', ntapes_onfile
              call endrun(msg=' ERROR: number of ntapes differs from restart file. '// &
                   'You can NOT change history options on restart.', &
                   additional_msg=errMsg(sourcefile, __LINE__))
           end if
 
-          if (ntapes > 0) then
-             allocate(history_tape_in_use_onfile(ntapes))
+          ntapes_gt_0: if (ntapes > 0) then
+             ! 4) TODO DONE Changed history_tape_in_use_onfile(t) to (t,f) throughout
+             allocate(history_tape_in_use_onfile(ntapes, maxsplitfiles))
              call ncd_io('history_tape_in_use', history_tape_in_use_onfile, 'read', ncid, &
                   readvar=readvar)
```

Comment:
As soon as I change `history_tape_in_use_onfile` to 2D, I get the
`Error: There is no specific subroutine for the generic 'ncd_io' at (1)`
My current first guess for why: This var is a boolean...

Reply:
Changed to integer and the code builds and runs!